x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/classification/MulticlassConfusionMatrix.java

```diff
@@ -0,0 +1,276 @@
+                        .otherBucket(true)
+                        .otherBucketKey(OTHER_BUCKET_KEY)));
+        } else {
+            return List.of();
+        }
+    }
+
+    @Override
+    public void process(Aggregations aggs) {
+        if (aggs.get(STEP_1_AGGREGATE_BY_ACTUAL_CLASS) != null && topActualClassNames == null) {
+            Terms termsAgg = aggs.get(STEP_1_AGGREGATE_BY_ACTUAL_CLASS);
+            topActualClassNames = termsAgg.getBuckets().stream().map(Terms.Bucket::getKeyAsString).collect(Collectors.toList());
+        }
+        if (aggs.get(STEP_2_AGGREGATE_BY_ACTUAL_CLASS) != null && result == null) {
+            Cardinality cardinalityAgg = aggs.get(STEP_2_CARDINALITY_OF_ACTUAL_CLASS);
+            Terms termsAgg = aggs.get(STEP_2_AGGREGATE_BY_ACTUAL_CLASS);
+            Map<String, Map<String, Long>> counts = new TreeMap<>();
+            for (Terms.Bucket bucket : termsAgg.getBuckets()) {
+                String actualClass = bucket.getKeyAsString();
+                Map<String, Long> subCounts = new TreeMap<>();
+                counts.put(actualClass, subCounts);
+                Filters subAgg = bucket.getAggregations().get(STEP_2_AGGREGATE_BY_PREDICTED_CLASS);
+                for (Filters.Bucket subBucket : subAgg.getBuckets()) {
+                    String predictedClass = subBucket.getKeyAsString();
+                    Long docCount = subBucket.getDocCount();
+                    if ((OTHER_BUCKET_KEY.equals(predictedClass) && docCount == 0L) == false) {
+                        subCounts.put(predictedClass, docCount);
+                    }
+                }
+            }
+            result = new Result(counts, termsAgg.getSumOfOtherDocCounts() == 0 ? 0 : cardinalityAgg.getValue() - size);
+        }
+    }
+
+    @Override
+    public Optional<EvaluationMetricResult> getResult() {
+        return Optional.ofNullable(result);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeVInt(size);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(SIZE.getPreferredName(), size);
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        MulticlassConfusionMatrix that = (MulticlassConfusionMatrix) o;
+        return Objects.equals(this.size, that.size);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(size);
+    }
+
+    public static class Result implements EvaluationMetricResult {
+
+        private static final ParseField CONFUSION_MATRIX = new ParseField("confusion_matrix");
+        private static final ParseField OTHER_CLASSES_COUNT = new ParseField("other_classes_count");
```

Comment:
I wonder if this should also be called just `_other_`.

Reply:
Done on the API level.

LMK if you think I should also rename the variables. In the code I would slightly prefer more descriptive "other_classes_count".